internals/overlord/overlord.go

```diff
@@ -57,6 +57,9 @@ var (
 
 	// Hold no more than 1000 completed notices, even if they have not yet expired
 	pruneMaxNotices = 1000
+
+	// Hold	no more than 100 serviceData.
```

Comment:
```suggestion
	// Hold no more than 100 inactive services.
```

Reply:
I see the issue now (which explains the discussion in the other comment), it was because I thought we would only keep 100 entries of serviceData, both active or not, that was why I was calculating how many inactive serviceData entries need to be pruned to reach this target.

This comment cleared things up. If it is the number of inactive services, rather than the total entries of serviceData, then it's more straightforward. I will make the changes accordingly, and probably need to update tests as well.